src/core/js/tracking.js

```diff
@@ -0,0 +1,111 @@
+define([
+    'core/js/adapt',
+    'core/js/enums/completionStateEnum'
+], function(Adapt, COMPLETION_STATE) {
+
+    var Tracking = Backbone.Controller.extend({
+
+        _config: {
+            _requireContentCompleted: true,
+            _requireAssessmentCompleted: false
+        },
+
+        _assessmentState: null,
+
+        initialize: function() {
+            Adapt.once('configModel:loadCourseData', this.loadConfig.bind(this));
+            Adapt.on('app:dataReady', this.setupEventListeners.bind(this));
+        },
+
+        setupEventListeners: function() {
+            // Check if completion requires passing an assessment. 
+            if (this._config._requireAssessmentCompleted) {
+                this.listenTo(Adapt, 'assessment:complete', this.onAssessmentComplete);
+            }
+
+            // Check if completion requires completing all content.
+            if (this._config._requireContentCompleted) {
+                this.listenTo(Adapt.course, 'change:_isComplete', this.checkCompletion);
+            }
+        },
+
+        /**
+         * Store the assessment state.
+         * @param {object} assessmentState - The object returned by Adapt.assessment.getState()
+         */
+        onAssessmentComplete: function(assessmentState) {
+            this._assessmentState = assessmentState;
+
+            this.checkCompletion();
+        },
+
+        /**
+         * Evaluate the course and assessment completion.
+         */
+        checkCompletion: function() {
+            var completionData = this.getCompletionData();
+
+            if (completionData.status === COMPLETION_STATE.INCOMPLETE) {
+                return;
+            }
+
+            Adapt.trigger('tracking:complete', completionData);
+            Adapt.log.debug('tracking:complete', completionData);
+        },
+
+        /**
+         * The return value of this function should be passed to the trigger of 'tracking:complete'.
+         * @returns An object representing the user's course completion.
+         */
+        getCompletionData: function() {
+            var completionData = {
+                status: COMPLETION_STATE.INCOMPLETE,
+                assessment: null
+            };
+
+            // Course complete is required.
+            if (this._config._requireCourseComplete && !Adapt.course.get('_isComplete')) {
```

Comment:
this needs changing to `_requireContentCompleted` as that's what's defined in the .schema

Reply:
Good catch.  I think the property name went through a few iterations.